IOHexperimenter/Cpp.md

```diff
@@ -72,52 +76,81 @@ const auto &suite_factory = ioh::suite::SuiteRegistry<ioh::problem::Real>::insta
+The default logger records evaluation information during the optimization process into csv files. An example of using the default logger is provided in [logger_example.h](https://github.com/IOHprofiler/IOHexperimenter/blob/master/example/logger_example.h)
+
+
+We declare a default logger using the following statement.
 ```cpp
-auto l = ioh::logger::Default("./", "logger_example", "random_search", "a random search for testing the bbob suite");
+auto logger = ioh::logger::Default(
+        fs::current_path(), /* output_directory : fs::current_path() -> working directory */
+        "folder_name", /* folder_name */
+        "random_search", /* algorithm_name : "random search'*/
+        "a random search" /* algorithm_info : "a random search for testing the bbob suite" */
+        );
 ```
 
-By default, evaluations when the best-found fitness is updated will be recorded into csv files. This can be changed by passing arguments to the constructor. In practice,
+We attach the `logger` to the `suite`, to perform logging process during optimization.
+```cpp
+suite->attach_logger(logger);
+```
+
+Alternatively, we can attach the logger to each new problem in a suite manually using
+```cpp
+problem->attach_logger(logger);
+```
+<a name = "tracking-parameters"></a>
+### Tracking Parameters
+The default loggers provides interfaces for tracking parameters of algorithms during optimization process. An example is provided in the [logger_example.h](https://github.com/IOHprofiler/IOHexperimenter/blob/da22d89fe1673ea67962829d12873e01387f6895/example/logger_example.h#L80)
 
-`t_always = true` turns on the .cdat files, which record all evaluations.
 
-`t_on_interval = 0` turns off the .idat files. If a positive value i is assigned, evaluations will be recorded in .idat files every i times,
+`experiment_attribute` stores the fixed parameters for each experiment, `run_attributes` maintains the parameters of each run of an algorithm, and `logged_attributes` records the online parameters for each evaluation.
+```cpp
+// Add parameters fixed throughout the experiment.
+  logger.add_experiment_attribute("meta_data_x", 69);
+  logger.add_experiment_attribute("meta_data_y", 69);
 
-`t_on_improvement = true` turns on the .dat files, which record evaluations when the best-found fitness is updated,
+  // Initialize parameters unique for each run. 
+  logger.create_run_attributes({"run_id"});
 
-`t_per_time_range = tp` and `t_at_time_points = ta` control the .tdat files together. The .tdat are turned off by default. Please find details in the [Sec 3. of the document](https://arxiv.org/pdf/1810.05281.pdf).
+  // Initialize parameters unique for each evaluation.
+  logger.create_logged_attributes({"x1"});
+```
```

Comment:
I would prefer not having x1 as the parameter, since the position can actually be stored automatically (which is also something we should document here).